### PR TITLE
feat(AIProvider): implement You service

### PR DESCRIPTION
**Pull request by dene-**
:roto2:

3739f6a1-7c66-4ceb-9cfb-a803e47ddf28

### CODE REVIEW

File: src/context/shared/application/ConfigService.ts [hunk 1]
```diff
@@ -107,6 +108,17 @@ export class OpenRouterConfig {
   model: string;
 }
 
+export class YouConfig {
+  @Property({
+    doc: `You.com models: src/context/AIProvider/infrastructure/You.const.ts`,
+    format: String,
+    env: 'OPENROUTER_MODEL',
+    nullable: process.env.AI_PROVIDER == openRouterProvider,
```
Comment (haskou) [inline src/context/shared/application/ConfigService.ts:116]: This should be compared with `youProvider` and not with `openRouterProvider`

Comment (dene-) [reply src/context/shared/application/ConfigService.ts]: Done.

File: src/context/AIProvider/domain/AIProvider.ts [hunk 2]
```diff
@@ -18,5 +18,5 @@ export interface AIProvider {
     isReplied?: boolean,
   ): Promise<string[]>;
 
-  generateConceptualHistory(memoryPrompt: string): Promise<string>;
+  generateConceptualHistory?(memoryPrompt: string): Promise<string>;
```
Comment (haskou) [inline src/context/AIProvider/domain/AIProvider.ts:21]: Don't modify Domain's interface, just throw a non implemented error

Comment (dene-) [reply src/context/AIProvider/domain/AIProvider.ts]: Done.

**Pull request merged**